### PR TITLE
migrates to version catalogs - needs some more cleanup

### PR DESCRIPTION
**Pull request by oneSIX**

### CODE REVIEW

**Comment by joshing-dev** [comment]
We'll need a rebase of this PR since main just took in the oauth functionality

File: build.gradle.kts [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-
-tasks.register("clean", Delete::class) {
-  delete(rootProject.buildDir)
+    alias(libs.plugins.android.application) apply false
+    alias(libs.plugins.android.library) apply false
+    alias(libs.plugins.kotlin.android) apply false
+    alias(libs.plugins.kotlin.multiplatform) apply false
+    alias(libs.plugins.kotlin.serialization) apply false
+    //alias(libs.plugins.kotlin.kapt) apply false
+    //alias(libs.plugins.ksp) apply false
```
Comment (joshing-dev) [inline build.gradle.kts:9]: @oneSIX I commented these out for now as they aren't currently used, do we want to keep them in/intend to use them?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+loginoauth = "0.11.1"
+multiplatform-settings = "1.0.0"
+timber = "5.0.1"
+
+[libraries]
+android-gradlePlugin = { group = "com.android.tools.build", name = "gradle", version.ref = "androidGradlePlugin" }
+androidx-activity-compose = { group = "androidx.activity", name = "activity-compose", version.ref = "androidxActivity" }
+
+# uses compose-bom so no version.ref for some deps
+androidx-compose-bom = { group = "androidx.compose", name = "compose-bom", version.ref = "androidxComposeBom" }
```
Comment (joshing-dev) [inline gradle/libs.versions.toml:28]: If we're using the BOM should we remove the `version.ref` for most of the compose dependencies below? @oneSIX

File: mastadonApp/build.gradle.kts [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
-  kotlin("android")
+  alias(libs.plugins.android.application)
+  alias(libs.plugins.kotlin.android)
+  //alias(libs.plugins.kotlin.kapt)
+  //alias(libs.plugins.ksp)
 }
-
 android {
   namespace = "com.matrix159.mastadonclone"
   compileSdk = 33
```
Comment (joshing-dev) [inline mastadonApp/build.gradle.kts:11]: @oneSIX Thoughts on also putting these in the TOML?

File: mastadonApp/build.gradle.kts [hunk 4]
```diff
@@ -30,23 +33,18 @@ android {
   }
   buildTypes {
     getByName("release") {
-      isMinifyEnabled = true
+      signingConfig = signingConfigs.getByName("debug")
```
Comment (joshing-dev) [inline mastadonApp/build.gradle.kts:38]: We probably don't want to keep it like this right? Maybe a TODO for when it comes time to actually making release builds for the play store?

File: presentation/build.gradle.kts [hunk 5]
```diff
@@ -40,32 +41,33 @@ android {
 }
 
 dependencies {
-  val composeVersion = "1.4.0-alpha04"
-  val coilVersion = "2.2.2"
+  val composeBom = platform(libs.androidx.compose.bom)
```
Comment (joshing-dev) [inline presentation/build.gradle.kts:42]: I added in these couple lines from the official documentation to make use of the BOM.

File: presentation/build.gradle.kts [hunk 6]
```diff
@@ -40,32 +41,33 @@ android {
...
+  debugImplementation(libs.androidx.compose.ui.tooling)
+  implementation(libs.androidx.compose.ui.tooling.preview)
 
-  implementation("androidx.activity:activity-compose:1.6.1")
-  implementation("androidx.lifecycle:lifecycle-runtime-compose:2.6.0-alpha04")
-  // Image loading
-  implementation("io.coil-kt:coil:$coilVersion")
-  implementation("io.coil-kt:coil-compose:$coilVersion")
+  // TODO remove this????
+  //implementation("androidx.compose.material:material:$composeVersion")
```
Comment (joshing-dev) [inline presentation/build.gradle.kts]: I think this was leftover remnants from my copy/pasted dependencies on my other project, and I *think* is fine to remove?

File: build.gradle.kts (hunk 1 above)
Comment (oneSIX) [reply build.gradle.kts:9]: I think this was just some leftover copy/pasta from when I was setting it up. We will probably end up using KSP later but for now i'll remove this.

File: gradle/libs.versions.toml (hunk 2 above)
Comment (oneSIX) [reply gradle/libs.versions.toml:28]: Done.

File: mastadonApp/build.gradle.kts (hunk 3 above)
Comment (oneSIX) [reply mastadonApp/build.gradle.kts:11]: For these constants I guess they're okay in this app module - I don't foresee there being more than one Android target?

File: presentation/build.gradle.kts (hunk 6 above)
Comment (oneSIX) [reply presentation/build.gradle.kts]: agreed.  I think we should migrate to just using Material3 as much as possible.

**Pull request merged**